src/Microsoft.SqlTools.ServiceLayer/EditData/EditSession.cs

```diff
@@ -150,6 +195,11 @@ public EditCreateRowResult CreateRow()
         /// <param name="errorHandler">Callback to perform if the commit process has failed at some point</param>
         public void CommitEdits(DbConnection connection, Func<Task> successHandler, Func<Exception, Task> errorHandler)
         {
+            if (!IsInitialized)
```

Comment:
Minor: since you check for this in every method, why not have a 1-line check by creating a `ThrowIfNotInitialized` method and calling it. We do this in other codebases as it shortens up methods.